### PR TITLE
feat:  Adds MongoDB storage implementation.

### PR DESCRIPTION
**Pull request by pfeairheller**
This PR included

- Storage implementation using MongoDB

Signed-off-by: Phil Feairheller <pfeairheller@gmail.com>

### CODE REVIEW

**Comment by troyronda** [comment]
@rolsonquadras volunteered to help review this PR.

File: component/storage/mongodb/mongodb.go [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+		return nil, errors.Wrap(err, "unable to create new mongo client opening store")
+	}
+
+	err = mongoClient.Connect(context.Background())
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to connect to mongo opening a new store")
+	}
+
+	client := mongoClient
+	db := client.Database(name)
```
Comment (rolsonquadras) [inline component/storage/mongodb/mongodb.go:85]: Do we need extra reassignment to `client` ?

File: component/storage/mongodb/mongodb.go [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+// Data is a wrapper for the stored key/value pair.
+type Data struct {
```
Comment (rolsonquadras) [inline component/storage/mongodb/mongodb.go:26]: Do we need to export this struct ? Looks like this struct is only for internal use.

**Review by rolsonquadras — APPROVED** [review]

**Review by troyronda — APPROVED** [review]

File: component/storage/mongodb/mongodb.go (hunk 1 above)
Comment (pfeairheller) [reply component/storage/mongodb/mongodb.go]: Absolutely do not, good catch!  Fixed.

File: component/storage/mongodb/mongodb.go (hunk 2 above)
Comment (pfeairheller) [reply component/storage/mongodb/mongodb.go]: Local makes more sense.  Fixed.

**Pull request merged**